api/client/build.go

```diff
@@ -308,13 +322,21 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		if jerr.Code == 0 {
 			jerr.Code = 1
 		}
+		if *suppressOutput {
+			fmt.Fprintln(cli.err, outBuf, sopts.out)
+		}
 		return Cli.StatusError{Status: jerr.Message, StatusCode: jerr.Code}
 	}
-
 	if err != nil {
 		return err
 	}
 
+	// Everything worked so if -q was provided the output from the daemon
+	// should be just the image ID and we'll print that to stdout.
+	if *suppressOutput {
+		fmt.Fprintln(cli.out, sopts.out)
```

Comment:
Wondering; will this output the *full* image ID? I noticed build currently outputs the truncated ID;

    Successfully built f0aeba98f104